File: tests.php5/BinaryData.java
public class BinaryData {

    public byte[] b;
    public byte[] getData(int len) {
	b = new byte[len];
	for(int i=0; i<b.length; i++) {
	    b[i]=(byte)(i%255);
	}
	return b;
    }

    public byte[] compare(byte[] b2) throws java.lang.Exception {
	if(b.length!=b2.length) throw new RuntimeException();
	for(int i=0; i<b2.length; i++)
	    if(b2[i]!=b[i]) throw new RuntimeException();
	return b2;
    }
    public String toString() throws java.lang.Exception {
	return new String(b, "ASCII");
    }
}

File: legacy/tests.php5/binaryData.php
<?php

include_once ("java/Java.inc");
ini_set("max_execution_time", 0);

$here=realpath(dirname($_SERVER["SCRIPT_FILENAME"]));
if(!$here) $here=getcwd();

$binaryData = new java("BinaryData");
$data = java_values($binaryData->getData(700*1024));
for($i=0; $i < 10; $i++) {
  $data=java_values($binaryData->compare($data));
  $str1=substr($data, 255, 256);
}
$str='&;a&amp;&quote"&quot;&&&;;"';
$binaryData->b='&;a&amp;&quote"&quot;&&&;;"';
$binaryData->compare('&;a&amp;&quote"&quot;&&&;;"');
if($str!=java_values($binaryData->toString())) { echo "ERROR\n"; exit(1); }

$data = java_values($binaryData->getData(1024));
if(strlen($data)!=1024) { echo "ERROR\n"; exit(5); }
$s1=substr(java_values($binaryData->toString()), 0, 256);
$binaryData->b=$str1;
$s2=substr(java_values($binaryData->toString()), 0, 256);

if($s1!=$s2) { echo "ERROR\n"; exit(2); }
echo "test ok\n";
exit(0);

?>
